src/include/duckdb/function/aggregate/nested_functions.hpp

```diff
@@ -13,11 +13,107 @@
 
 namespace duckdb {
 
-struct ListFun {
-	static void RegisterFunction(BuiltinFunctions &set);
-};
 struct HistogramFun {
 	static void RegisterFunction(BuiltinFunctions &set);
 	static AggregateFunction GetHistogramUnorderedMap(LogicalType &type);
 };
+
+struct ListSegment {
+	uint16_t count;
+	uint16_t capacity;
+	ListSegment *next;
+};
+struct LinkedList {
+	LinkedList() {};
+	LinkedList(idx_t total_capacity_p, ListSegment *first_segment_p, ListSegment *last_segment_p)
+	    : total_capacity(total_capacity_p), first_segment(first_segment_p), last_segment(last_segment_p) {
+	}
+
+	idx_t total_capacity = 0;
+	ListSegment *first_segment = nullptr;
+	ListSegment *last_segment = nullptr;
+};
+struct ListFun {
+	static void RegisterFunction(BuiltinFunctions &set);
```

Comment:
Do these structs and functions need to be in the `.hpp` files?

If they are only used in `list.cpp` I'd prefer moving them there, then the functions also don't need to be members of `ListFun`.